src/sentry/integrations/github/integration.py

```diff
@@ -808,23 +824,64 @@ def dispatch(self, request: HttpRequest, pipeline: Pipeline) -> HttpResponseBase
                 lifecycle.record_failure(GitHubInstallationError.MISSING_TOKEN)
                 return error(
                     request,
-                    self.active_organization,
+                    self.active_user_organization,
                     error_short=GitHubInstallationError.MISSING_TOKEN,
                 )
-            client = GithubSetupApiClient(access_token=payload["access_token"])
-            authenticated_user_info = client.get_user_info()
+            self.client = GithubSetupApiClient(access_token=payload["access_token"])
+            authenticated_user_info = self.client.get_user_info()
+
+            if self.active_user_organization is not None and features.has(
+                "organizations:github-multi-org",
+                organization=self.active_user_organization.organization,
+                actor=request.user,
+            ):
+                owner_orgs = self._get_owner_github_organizations()
+
+                installation_info = self._get_eligible_multi_org_installations(
+                    owner_orgs=owner_orgs
+                )
+                pipeline.bind_state("existing_installation_info", installation_info)
 
             if "login" not in authenticated_user_info:
                 lifecycle.record_failure(GitHubInstallationError.MISSING_LOGIN)
                 return error(
                     request,
-                    self.active_organization,
+                    self.active_user_organization,
                     error_short=GitHubInstallationError.MISSING_LOGIN,
                 )
-
             pipeline.bind_state("github_authenticated_user", authenticated_user_info["login"])
             return pipeline.next_step()
 
+    def _get_owner_github_organizations(self) -> list[str]:
+        user_org_membership_details = self.client.get_organization_memberships_for_user()
+
+        return [
+            gh_org.get("organization", {}).get("login")
+            for gh_org in user_org_membership_details
+            if (
+                gh_org.get("role", "").lower() == "admin"
+                and gh_org.get("state", "").lower() == "active"
+            )
+        ]
+
+    def _get_eligible_multi_org_installations(
+        self, owner_orgs: list[str]
+    ) -> list[GithubInstallationInfo]:
+        installed_orgs = self.client.get_user_info_installations()
+
+        return [
```

Comment:
We use the list of orgs the user is an owner from above to determine which organizations (or user accounts) where the Sentry GH app is already installed on should be available to be selected from.